### DISCUSSION TITLE
I have a question about configuring NAS and ZFS.

### DISCUSSION THREAD

**Discussion by ghost** [minio/minio#14594] [Q&A] [answered]
Hello
I'm trying to build a MinIO server.
However, MinIO recommends XFS and JBOD methods.
The configuration I am going to do is to connect the NAS to the Linux server through iSCSI and connect the ZFS file system to the configuration.
Does the above configuration work well?
And is there any reason to recommend the XFS and JBOD methods?

**Comment by klauspost** [comment, marked as answer]
Does it work: Probably
Does it work well: Depends on what your definition of "well" is.
Would we recommend it: No.

JBOD and XFS is faster, more reliable. You are welcome to do your own tests of the system. If it performs to your satisfaction, go ahead, but if it doesn't, you will know why.

**Reply by ghost** [reply]
Thank you for your reply.

I'm sorry, but I'd like to ask you one more question.

4 Ubuntu servers.
- cpu : Intel(R) Xeon(R) Silver 4208 CPU @ 2.10GHz
- RAM : 256GB
- network : 10G

4 NAS servers
- 7.3TB HDD*12 (RAID 5 configuration)
- network : 10G

Through the above environment, I have configured MinIO.

1. Configure NAS disks as one LUN in RAID5
2. Attach NAS's LUN to Ubuntu server via iSCSI
3. Mount the LUN attached to the Ubuntu server by ZFS
4. ./minio server https://minio-{1...4}/volume1/data{1...4}
Execute (create 4 directories and then connect)

Data transfer to rclone at test time (over 4.7TB capacity, 25,711,571 objects)

There was a deterioration in performance during the test.
Usually I/O wa stays 0-5 but increases to wa 80 when bucket reloads on MinIO web.

And during the test, more than half of the stored data disappeared when one server failed and went down.

Is the performance insufficient when constructing the above?
I set the parity to 4, but more than half of the data disappeared in case of failure. Why?

**Reply by harshavardhana** [reply]
Don't use NAS with MinIO - MinIO is already a NAS - no reason to do erasure coding on top of erasure coding. This is not supported.

**Reply by ghost** [reply]
Thank you for your reply.

Will using file system ZFS affect performance?

**Reply by harshavardhana** [reply]
Well TBH we have no idea, we never recommend any specialized filesystems underneath. We expect that you JBODs to MinIO and Put XFS on it.

**Reply by ghost** [reply]
Thank you.
It was very helpful.
